presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveStorageFormats.java

```diff
@@ -1,8 +1,284 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.facebook.presto.tests.hive;
 
-/**
- * Created by andrii on 21.03.16.
- */
+import com.facebook.presto.jdbc.PrestoConnection;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.assertions.QueryAssert.Row;
+import com.teradata.tempto.query.QueryResult;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.tests.TestGroups.STORAGE_FORMATS;
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableList;
+import static com.teradata.tempto.assertions.QueryAssert.Row.row;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.query.QueryExecutor.defaultQueryExecutor;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
 public class TestHiveStorageFormats
+        extends ProductTest
 {
+    private static final String TPCH_SCHEMA = "tiny";
+
+    @DataProvider(name = "storage_formats")
+    public static Object[][] storageFormats()
+    {
+        return new StorageFormat[][] {
+                //{storageFormat("ORC")},
+                //{storageFormat("DWRF")},
+                //{storageFormat("PARQUET")},
+                //{storageFormat("PARQUET", ImmutableMap.of("hive.parquet_optimized_reader_enabled", "true"))},
+                //{storageFormat("RCBINARY")},
+                //{storageFormat("RCTEXT")},
+                //{storageFormat("SEQUENCEFILE")},
+                //{storageFormat("TEXTFILE")}
+        };
+    }
+
+    @Test(dataProvider = "storage_formats", groups = {STORAGE_FORMATS})
+    public void testInsertIntoTable(StorageFormat storageFormat)
+    {
+        setSessionProperties(storageFormat);
+
+        String tableName = "storage_formats_test_insert_into_" + storageFormat.getName().toLowerCase();
+
+        // DROP TABLE
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+
+        // CREATE TABLE
+        String createTable = format(
+                "CREATE TABLE %s(" +
```

Comment:
It will also help as the types of these columns change when `VARCHAR(x)` is complete and `DECIMAL` is added